packages/react/src/primitives/ToggleButtonGroup/useToggleButtonGroup.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback } from 'react';
+
+import { ToggleButtonProps, ToggleButtonGroupProps } from '../types';
+export const useToggleButtonGroup = (
+  onChange: ToggleButtonGroupProps['onChange'],
+  value: ToggleButtonGroupProps['value'],
+  isExclusive = false
+) => {
+  // Multiple selection
+  const handleChange: ToggleButtonProps['onChange'] = useCallback(
+    (e, buttonValue) => {
+      if (!onChange || !Array.isArray(value)) {
+        return;
+      }
+
+      const index = value.indexOf(buttonValue);
+      let newValue;
+
```

Comment:
non-blocking suggestion: separating this logic into named variable might increase readability:

```suggestion
const shouldToggleOff = index >= 0;

// then later:
if (shouldToggleOff) {
```